src/routes/tradesToPublish.ts

```diff
@@ -0,0 +1,102 @@
+import { Request } from "express";
+import { Response } from "express";
+import { getLogger } from "log4js";
+import { BitcoinConfig } from "../config";
+import Ledger from "../ledger";
+import { Trade, TradeService } from "../rates/tradeService";
+import { EthereumWallet } from "../wallets/ethereum";
+
+const logger = getLogger();
+
+const previousTradesLookup: Map<string, Trade> = new Map<string, Trade>();
+
+export function getAmountsToPublishRoute(
+  tradeService: TradeService,
+  bitcoinConfig: BitcoinConfig,
+  ethereumWallet: EthereumWallet,
+  peerId: string,
+  addressHint?: string
+) {
+  // @ts-ignore
+  return async (req: Request, res: Response) => {
+    try {
+      if (!bitcoinConfig) {
+        throw new Error("Bitcoin not configured correctly.");
+      }
+      if (!ethereumWallet) {
+        throw new Error("Ethereum not configured correctly");
+      }
+
+      const trades = await tradeService.prepareTradesToPublish();
+
+      const publishTradesResponse = {
+        peerId,
+        addressHint,
+        ledgers: [
+          {
+            name: Ledger.Bitcoin,
+            network: bitcoinConfig.network
+          },
+          {
+            name: Ledger.Ethereum,
+            network: ethereumWallet.getChainId() === 17 ? "regtest" : "testnet" // TODO: refactor Ethereum ChainId handling in bobtimus
+          }
+        ],
+        rates: trades.map(trade => {
+          return {
+            timestamp: formatTimestamp(findLatestTradeTimestamp(trade)),
```

Comment:
This is what I still find strange, the timestamp is stored separately to the trade.